tests/Database/DatabaseEloquentCollectionTest.php

```diff
@@ -229,6 +230,18 @@ public function testWithHiddenSetsHiddenOnEntireCollection()
 
         $this->assertEquals([], $c[0]->getHidden());
     }
+
+    public function testNonModelRelatedMethods()
+    {
+        $a = new Collection([['foo' => 'bar'], ['foo' => 'baz']]);
+        $b = new Collection(['a', 'b', 'c']);
+        $this->assertInstanceOf(BaseCollection::class, $a->pluck('foo'));
+        $this->assertInstanceOf(BaseCollection::class, $a->keys());
+        $this->assertInstanceOf(BaseCollection::class, $a->collapse());
+        $this->assertInstanceOf(BaseCollection::class, $a->flatten());
+        $this->assertInstanceOf(BaseCollection::class, $a->zip(['a', 'b'], ['c', 'd']));
+        $this->assertInstanceOf(BaseCollection::class, $b->flip());
```

Comment:
These don't really test what you want.

An Eloquent collection is an instance of the base collection.